conan/tools/gnu/autotoolstoolchain.py

```diff
@@ -291,3 +296,24 @@ def generate_args(self):
                 "make_args":  cmd_args_to_string(self.make_args),
                 "autoreconf_args": cmd_args_to_string(self.autoreconf_args)}
         save_toolchain_args(args, namespace=self._namespace)
+
+
+class _CustomCrossBuildUX:
+    def __init__(self, arch, is_host, configure_args):
+        self.arch = arch
+        self.is_host = is_host
+        self.configure_args = configure_args
+
+    def __getitem__(self, item):
+        if item == "arch":
+            return self.arch
+        elif item == "triplet":
+            return self.triplet
+
+    def __setitem__(self, key, value):
+        if key == "arch":
+            self.arch = value
+        elif key == "triplet":
+            flag = "host" if self.is_host else "build"
+            self.triplet = value
+            self.configure_args.extend([f"--{flag}={value}"])
```

Comment:
I don't know if it is a good idea the set directly introduce the value to configure_args